src/plugins/vis_type_vega/public/line_vega_spec_expression.ts

```diff
@@ -0,0 +1,246 @@
+  // @ts-ignore
+  endTime[xAxisId] = new Date(dimensions.x.params.bounds.max).valueOf();
+  const updatedTable = datatable.rows.concat([startTime, endTime]);
+
+  // TODO: update this to v5 when available
+  spec.$schema = 'https://vega.github.io/schema/vega-lite/v4.json';
+  spec.data = {
+    values: updatedTable,
+  };
+  spec.config = {
+    view: {
+      stroke: null,
+    },
+    concat: {
+      spacing: 0,
+    },
+    // the circle timeline representing annotations
+    circle: {
+      color: 'blue',
+    },
+    // the vertical line when user hovers over an annotation circle
+    rule: {
+      color: 'red',
+    },
+    legend: {
+      orient: legendPosition,
+    },
+  };
+
+  // Get the valueAxes data and generate a map to easily fetch the different valueAxes data
+  const valueAxis = {};
+  parseParams.valueAxes.forEach((yAxis: { id: { toString: () => string | number } }) => {
+    // @ts-ignore
+    valueAxis[yAxis.id.toString()] = yAxis;
+  });
+
+  spec.layer = [] as any[];
+
+  if (datatable.rows.length > 0) {
+    let skip = 0;
+    datatable.columns.forEach((column, index) => {
+      const currentSeriesParams = parseParams.seriesParams[index - skip];
+      // Check if its not xAxis column data
+      if (column.meta?.aggConfigParams?.interval != null) {
+        skip++;
+      } else {
+        const currentValueAxis =
+          // @ts-ignore
+          valueAxis[currentSeriesParams.valueAxis.toString()];
+        let tooltip: Array<{ field: string; type: string; title: string }> = [];
+        if (parseParams.addTooltip) {
+          tooltip = [
+            { field: xAxis.id, type: 'temporal', title: xAxis.name.replaceAll('"', '') },
+            { field: column.id, type: 'quantitative', title: column.name.replaceAll('"', '') },
+          ];
+        }
+        spec.layer.push({
+          mark: {
+            type: currentSeriesParams.type,
+            interpolate: currentSeriesParams.interpolate,
+            strokeWidth: currentSeriesParams.lineWidth,
+            point: currentSeriesParams.showCircles,
+          },
+          encoding: {
+            x: {
+              axis: {
+                title: xAxis.name.replaceAll('"', ''),
+                grid: parseParams.grid.categoryLines,
+              },
+              field: xAxis.id,
+              type: 'temporal',
+            },
+            y: {
+              axis: {
+                title:
+                  currentValueAxis.title.text.replaceAll('"', '') ||
+                  column.name.replaceAll('"', ''),
+                grid: parseParams.grid.valueAxis !== '',
+                orient: currentValueAxis.position,
+                labels: currentValueAxis.labels.show,
+                labelAngle: currentValueAxis.labels.rotate,
+              },
+              field: column.id,
+              type: 'quantitative',
+            },
+            tooltip,
+            color: {
+              datum: column.name.replaceAll('"', ''),
+            },
```

Comment:
How does this work exactly?

Reply:
I will add a comment here to clarify the `replaceAll`. I need to remove double quotation marks because if there is `"` in the name, it would cause vega to not render as it cannot handle taking `"`.

Reply:
Is `datum` a valid param for `color`?

Reply:
If we need to do escaping for datatable string names for use in the vega spec, we should just do that as part of the initial table prep. We could have a function called `formatTable` or something to do any transformations necessary before we start building the spec.

Reply:
> Is `datum` a valid param for `color`?

Yes it is. It used to make sure each different set of y values have their own color.